Sources/Automerge/Document.swift

```diff
@@ -16,27 +16,33 @@ public final class Document: @unchecked Sendable {
     private var doc: WrappedDoc
 
     #if !os(WASI)
-    fileprivate let queue = DispatchQueue(label: "automerge-sync-queue", qos: .userInteractive)
-    fileprivate func sync<T>(execute work: () throws -> T) rethrows -> T {
-        try queue.sync(execute: work)
+    let lock = NSRecursiveLock()
+    fileprivate func lock<T>(execute work: () throws -> T) rethrows -> T {
+        lock.lock()
+        defer { lock.unlock() }
+        return try work()
     }
     #else
-    fileprivate func sync<T>(execute work: () throws -> T) rethrows -> T {
+    fileprivate func lock<T>(execute work: () throws -> T) rethrows -> T {
         try work()
     }
     #endif
+    
+    #if canImport(Combine)
+    public let objectDidChange: PassthroughSubject<(), Never> = .init()
```

Comment:
Can you add a bit of doc comment here highlighting that this is triggered explicitly _after_ the document has been updated, and use your use-case as a "for example"? i.e. "Sends a signal that the document has been updated. For example, you can use this signal to retrieve `heads` for the document to capture the state after you applied changes." - or something like that?

Reply:
Yes, but I'm just in the middle of a mad summer dash, so won't get to it today, but will do.

Reply:
Jesse - I'll go ahead and merge this and add the docs after.